code/fireball/fireballs.cpp

```diff
@@ -373,6 +373,90 @@ static void parse_fireball_tbl(const char *table_filename)
 				// rather than just having the default model that might or might not be used
 				fi->use_3d_warp = true;
 			}
+
+			if (optional_string("$Bobboau anim:")) {
+				stuff_boolean(&fi->bobboau_anim);
+			} else {
+				fi->bobboau_anim = false;
```

Comment:
This is a common footgun to how our table parsing works. Consider the scenario of modular tables. If the parent table sets bobboau_anim to true and then a modular table omits the entry entirely then it will be set to false here in the else statement. It looks like fireball.tbl sets defaults for new fireballs around line 306 instead.

This comment applies to each place below where you parse with `optional_string()` followed by an `else` statement.